packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -693,6 +737,59 @@ function commitRoot(root: FiberRoot, finishedWork: Fiber): void {
     legacyErrorBoundariesThatAlreadyFailed = null;
   }
   onCommit(root, earliestRemainingTimeAfterCommit);
+
+  if (enableInteractionTracking) {
+    ((__interactionsRef: any): InteractionsRef).current = ((prevInteractions: any): Set<
+      Interaction,
+    >);
+
+    if (enableInteractionTrackingObserver) {
+      let caughtError = null;
+      let subscriber;
+
+      try {
+        subscriber = ((__subscriberRef: any): SubscriberRef).current;
+        if (
+          subscriber !== null &&
+          ((root.memoizedInteractions: any): Set<Interaction>).size > 0
+        ) {
+          // Interaction threads are unique per root and expiration time.
+          // DO NOT CHANGE THIS VALUE without also updating other threadID values.
+          const threadID =
+            committedExpirationTime * 1000 + root.interactionThreadID;
+          subscriber.onWorkStopped(
+            ((root.memoizedInteractions: any): Set<Interaction>),
+            threadID,
+          );
+        }
+      } catch (error) {
+        caughtError = error;
+      } finally {
+        // Now that we're done, check the completed batch of interactions.
+        // If no more work is outstanding for a given interaction,
+        // We need to notify the subscribers that it's finished.
+        ((committedInteractions: any): Array<Interaction>).forEach(
+          interaction => {
+            interaction.__count = ((interaction.__count: any): number) - 1;
+            if (
+              subscriber !== null &&
+              ((interaction.__count: any): number) === 0
+            ) {
+              try {
+                subscriber.onInteractionScheduledWorkCompleted(interaction);
+              } catch (error) {
+                caughtError = caughtError || error;
+              }
+            }
+          },
+        );
+
+        if (caughtError) {
+          throw caughtError;
```

Comment:
try/finally will just rethrow the error by default. You can remove the catch and this line.